### ISSUE TITLE
Upgrade and restoration failed

### ISSUE THREAD

**Issue by fflorent** [YunoHost-Apps/wekan_ynh#104]
**How to post a meaningful bug report**
1. *Read this whole template first.*
2. *Determine if you are on the right place:*
   - *If you were performing an action on the app from the webadmin or the CLI (install, update, backup, restore, change url...), you are on the right place!*
   - *Otherwise, the issue may be due to Wekan itself. Refer to its documentation or repository for help.*
   - *If you have a doubt, post here, we will figure it out together.*
3. *Delete the italic comments as you write over them below, and remove this guide.*
--- 

**Describe the bug**
I attempted to upgrade from version 4.09 to version 4.43~ynh1:
 - the upgrade failed
 - the restoration failed

**Versions**
- YunoHost version: 4.0.8
- I have access to my server: *Through SSH and through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance ?: *yes*
  - If yes, please explain: I tweaked settings like the mail configuration, but nothing that should interfere here (at least I think). I don't think I have tweaked anything related to wekan.
- Using, or trying to install package version/branch: 4.43~ynh1
- If upgrading, current package version: 4.09

**To Reproduce**
*Steps to reproduce the behavior.*
- *If you used the webadmin, please perform the equivalent command from the CLI first.*: I can reproduce the upgrade using the command line upon request
- *If the error occurs in your browser, explain what you did:*
   1. Go to this page: https://zeteo.me/yunohost/admin/#/update
   2. Click on the button to upgrade wekan only
   3. Wait for the upgrade and see it fail

Also 
**Logs**
https://paste.yunohost.org/raw/kafalaximo

---

I tried to restore a backup, but it also failed: https://paste.yunohost.org/raw/ibimasubod

I would really appreciate a workaround as I cannot even reinstall the backup. Thanks in advance!

**Comment by nukedone** [comment]
the same error here. Seems that the instance cannot connect to MongoDB

**Comment by anmol26s** [comment]
Do you have other apps using Mongodb?

What is ?
`$ mongo --version`

You should try fixing packages by:

`$ apt --fix-broken install`

**Comment by nukedone** [comment]
The error is because MongoDB service is not running. 
My issue, after installing MongoDB as in [here](https://linuxize.com/post/how-to-install-mongodb-on-debian-10/) up to step 4 didn't make the service start. 
Since there wasn't a service mongod.service I ran
```
sudo systemctl daemon-reload
```
And at the end had to remove 
```
sudo rm -rf /tmp/mongodb-27017.sock
```

**Comment by anmol26s** [comment]
Try removing the Mongodb.

`$ sudo apt-get purge mongodb-org*`

Then again installing it.
`$ sudo apt-get update && sudo apt-get install mongodb`

**Comment by stilobique** [comment]
Hi, same issue for me. I'm try to install Mongo DB and Yunohost give me an precious information : 
```
Aucune version du paquet mongodb n'est disponible, mais il existe dans la base
de données. Cela signifie en général que le paquet est manquant, qu'il est devenu obsolète
ou qu'il n'est disponible que sur une autre source
```

(No package are available in English :) .

**Comment by yalh76** [comment]
> 
> 
> I would really appreciate a workaround as I cannot even reinstall the backup. Thanks in advance!

Did you first install wekan on yunohost 3.x/Stretch, then at one time, upgrade yunohost to 4.x/Buster, and now tryed to upgrade wekan?

I'm asking that because mongodb has been removed from Debian in Buster and mongodb package name changed, with stretch it was mongodb and now, it's mongodb-org.

So maybe wekan upgrade doesn't manage that mongodb dependency name change...

**Comment by fflorent** [comment]
> Did you first install wekan on yunohost 3.x/Stretch, then at one time, upgrade yunohost to 4.x/Buster, and now tryed to upgrade wekan?

Yes, that's indeed my case.

> I'm asking that because mongodb has been removed from Debian in Buster and mongodb package name changed, with stretch it was mongodb and now, it's mongodb-org.

OK, for the workaround, may I install mongodb-org?

> So maybe wekan upgrade doesn't manage that mongodb dependency name change...

I guess… Do you think it can be easily tested? I don't have much time but if it can be checked in about 1h, I can make that effort.

Florent

**Comment by SiM** [comment]
Same problem here on several servers.

I found a "workaround" after some tries (:warning: do at your own risk, not sure if it will works in all contexts)
(I consider starting after a failed wekan upgrade)
* Purge old mongodb data (:warning: do a backup before and make sure wekan is the only app to use mongoDB)
`rm -rf /var/lib/mongodb`

* Install mongodb (cf. https://docs.mongodb.com/manual/tutorial/install-mongodb-on-debian/) : 
```
wget -qO - https://www.mongodb.org/static/pgp/server-4.4.asc | sudo apt-key add -
echo "deb http://repo.mongodb.org/apt/debian buster/mongodb-org/4.4 main" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.4.list
apt-get update
apt-get install -y mongodb-org
```

* Restore wekan backup made before the failed upgrade : 
`yunohost backup restore wekan-pre-upgradeX` (replace X with the correct number ; type `yunohost backup list` to know)

* Upgrade wekan
`yunohost app upgrade wekan`

---
It would be ideal if all this could be managed during the wekan upgrade process, but my skills are not high enough :smile:

**Comment by stilobique** [comment]
I'm try your comment @SiM and it's work to me.

**Comment by jlnbdt** [comment]
Thanks @SiM, it worked for me too :-)

**Comment by SiM** [comment]
After following the "workaround" described above, If you have a warning on YNH diagnosis system regarding mongodb service (it's the old one, the new one is mongod) : 
`yunohost service remove mongodb` will fix it

**Comment by JRelland** [comment]
Hum, 
I should have come here before opening a post on the YNH forum :( 
https://forum.yunohost.org/t/pb-avec-wekan-et-mongodb/14762/3

Thank you for your comment.

Same situation after an update, two days ago. 

Try several things mention here. Same problem. 

Let me try again... just because I love wekan :).

**Comment by jrd10** [comment]
Hi, 
sorry but the "workaround" does not work for my configuration.

My  `/etc/apt/trusted.gpg.d/wekan.asc' wasn't not deleted because it doesn't exist` is back! :(

Log here: https://paste.yunohost.org/raw/exoqegirec

Thank you for your time. jrd10

**Comment by jrd10** [comment]
And always in troubles with MongoDB

Although
```
Description   MongoDB daemon
Status    running since 1 second ago
Start on boot   Enabled
Configuration   unknown
```
Diagnosis says: `Service mongod is failed :( `  and make the `MongoDB daemon`

https://paste.yunohost.org/iheracenac

Thank you for your time.

**Comment by nukedone** [comment]
Try with `netstat -tulpn` if the port is not taken by another service. Also I remember that I had to manually add repository and install mongodb.

⁣Get BlueMail for Android ​

**Comment by yalh76** [comment]
should be solved for now
